internal/app/handle.go

```diff
@@ -0,0 +1,87 @@
+package app
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+type fighter struct {
+	Name string `json:"name"`
+}
+
+// struct to hold the new concatination of the user's name
+type allegiance struct {
+	Message string `json:"message"`
+}
+
+// gokuHandler accepts a post request that takes the user's name from the JSON request
+// and responds back with that user's name + some message
+func gokuHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
let's make the naming of this functions and case functions more idiomatic. functions should state what they do, the `-er` naming is better for an object/struct with methods. for functions they should be actions. so `handle` and since it handles all `/goku` requests, it can be `handleGokuRequests`. then for each rest case, `handleGokuPost` or `handleGokuGet` etc.